src/components/ReportActionItem/IOUPreview.js

```diff
@@ -136,32 +130,32 @@ const IOUPreview = (props) => {
     if (props.iouReport.total === 0) {
         return null;
     }
-
     const sessionEmail = lodashGet(props.session, 'email', null);
     const managerEmail = props.iouReport.managerEmail || '';
     const ownerEmail = props.iouReport.ownerEmail || '';
+    const participantEmails = props.isBillSplit ? props.action.originalMessage.participants : [managerEmail, ownerEmail];
+    const participantAvatars = OptionsListUtils.getAvatarsForLogins(participantEmails);
 
     // Pay button should only be visible to the manager of the report.
     const isCurrentUserManager = managerEmail === sessionEmail;
 
-    const managerName = ReportUtils.getDisplayNameForParticipant(managerEmail, true);
-    const ownerName = ReportUtils.getDisplayNameForParticipant(ownerEmail, true);
-    const managerAvatar = {
-        source: ReportUtils.getAvatar(lodashGet(props.personalDetails, [managerEmail, 'avatar']), managerEmail),
-        type: CONST.ICON_TYPE_AVATAR,
-        name: managerEmail,
-    };
-    const ownerAvatar = {
-        source: ReportUtils.getAvatar(lodashGet(props.personalDetails, [ownerEmail, 'avatar']), ownerEmail),
-        type: CONST.ICON_TYPE_AVATAR,
-        name: ownerEmail,
-    };
-    const cachedTotal = props.iouReport.total && props.iouReport.currency
-        ? props.numberFormat(
-            Math.abs(props.iouReport.total) / 100,
-            {style: 'currency', currency: props.iouReport.currency},
-        ) : '';
-    const avatarTooltip = [Str.removeSMSDomain(managerEmail), Str.removeSMSDomain(ownerEmail)];
+    let cachedTotal;
+    if (props.isBillSplit) {
+        if (props.action && props.action.originalMessage && props.action.originalMessage.currency) {
+            cachedTotal = props.numberFormat(
+                Math.abs(props.action.originalMessage.amount) / 100,
+                {style: 'currency', currency: props.action.originalMessage.currency},
+            );
+        } else {
+            cachedTotal = props.numberFormat(Math.abs(props.action.originalMessage.amount) / 100);
+        }
+    } else {
+        cachedTotal = props.iouReport.total && props.iouReport.currency
+            ? props.numberFormat(
+                Math.abs(props.iouReport.total) / 100,
+                {style: 'currency', currency: props.iouReport.currency},
+            ) : '';
+    }
```

Comment:
Can you leave a comment explaining why we get the amount from different sources depending on the type of IOU? For example, I don't exactly what's the difference between `props.iouReport.total` and `props.action.originalMessage.amount`.

Reply:
Ohh I see later that you left a comment saying "// Only 1:1 requests require an iouReport. For splits we rely on reportAction data.", I think you leave a comment saying the same here.

Reply:
Either way, instead of this complex if/else in which the code is almost repeated, I think you can simply find the proper total depending on the request type and then maybe call `props.numberFormat` once with the amount.

Reply:
Agree with only calling numberFormat once

Reply:
Thanks both, I vastly improved and simplified this logic.